packages/base-plugin/lib/plugin.js

```diff
@@ -1,68 +1,53 @@
-  //   '/session/:sessionId/new_method': {
-  //      GET: {command: 'getNewThing'},
-  //      POST: {command: 'setNewThing', payloadParams: {required: ['someParam']}}
-  //   }
-  // }
+/**
+ * @implements {Plugin}
+ */
+class BasePlugin {
   static newMethodMap = {};
 
-  constructor(pluginName, opts = {}) {
+  /** @type {Plugin['cliArgs']} */
+  cliArgs;
+
+  /**
+   * @param {string} pluginName
+   */
+  constructor(pluginName) {
     this.name = pluginName;
     this.logger = logger.getLogger(`Plugin [${pluginName}]`);
-
-    // allow for args to be used by plugins
-    this.opts = opts;
   }
 
   /**
    * Optionally updates an Appium express app and http server, by calling methods that may mutate
    * those objects. For example, you could call:
    *
-   * expressApp.get('/foo', handler)
+   * `expressApp.get('/foo', handler)`
    *
    * In order to add a new route to Appium with this plugin. Or, you could add new listeners to the
    * httpServer object.
    *
-   * @param {object} expressApp - the Express 'app' object used by Appium for route handling
-   * @param {http.Server} httpServer - the node HTTP server that hosts the app
-   */
-  /*static async updateServer (expressApp, httpServer) {
-  }*/
-
-  /**
-   * Handle an Appium command, optionally running and using or throwing away the value of the
-   * original Appium behavior (or the behavior of the next plugin in a plugin chain).
-   *
-   * @param {function} next - a reference to an async function which encapsulates what would
-   * normally happen if this plugin were not handling a command. If this is the only plugin
-   * handling the command, `await next()` would therefore trigger the normal handling logic in the
-   * driver which is in use. If another plugin is registered, it would run *that* plugin's `handle`
-   * method and return the result for use here. Note that if this plugin does *not* call
-   * `await next()`, then the normal command logic will not be run, and this plugin is responsible
-   * for managing new command timeouts and command logging, for example:
-   *   `driver.stopNewCommandTimeout()` -- before running plugin logic
-   *   `driver.startNewCommandTimeout()` -- after running plugin logic
-   *   `driver._eventHistory.commands.push({cmd: cmdName, startTime, endTime}) -- after running
-   *   plugin logic
-   * @param {BaseDriver} driver - the instance of the Appium driver currently handling commands
-   * @param {...object} args - the args that would be applied to the normal command
-   *
-   * @return {object} - the result to pass to the user
+   * @param {import('express').Express} expressApp - the Express 'app' object used by Appium for route handling
+   * @param {import('@appium/types').AppiumServer} httpServer - the node HTTP server that hosts the app
    */
-  /*async <cmdName> (next, driver, ...args) {
-    return await next();
-  }*/
+  // eslint-disable-next-line no-unused-vars
+  static async updateServer(expressApp, httpServer) {}
```

Comment:
since this is now defined, does it trigger the server update logic when loading plugins? we probably don't want it to do that unless plugins implement this?

Reply:
I don't know.  I'll find out.

Reply:
ok. that was the reason i originally left it commented out--the plugin loading harness checks for the existence of the function and calls it if it exists; it seemed safer to do it that way so we can explicitly log when a plugin is updating the server in this way to the user. with your change i worry that we wouldn't be able to know when a plugin actually is or isn't doing anything to the server.

Reply:
OK.  I've just removed `updateServer` (and `handle`‚ from `BasePlugin`, since a) it's a no-op, and b) it's not necessary.  Any useful comments have been moved into `packages/types/lib/plugin.ts`.